server/server/src/cors.rs

```diff
@@ -21,21 +21,29 @@ pub fn cors_policy(config_settings: &Settings) -> Cors {
         }
         cors = cors.allowed_origin_fn(|_header, req| {
             //allow requests where Sec-Fetch-Site is set to same-origin, same-site or none
-            let sec_fetch_site_header = req
-                .headers
-                .get("Sec-Fetch-Site")
-                .map(header::HeaderValue::as_bytes);
+            let sec_fetch_site_header = req.headers.iter().find_map(|(name, value)| {
+                (name.to_string().to_lowercase() == "sec-fetch-site").then(|| value.as_bytes())
+            });
 
             match sec_fetch_site_header {
-                None => false,
-                Some(bytes) => match bytes {
-                    b"cross-site" => false,
-                    b"same-origin" => true,
-                    b"same-site" => true,
-                    b"none" => true,
-                    _ => false,
-                },
-            }
+                Some(b"cross-site") => return true,
+                Some(b"same-origin") => return true,
+                Some(b"none") => return true,
+                _ => {}
+            };
+
+            // Allow for android and electron
+            // TODO restrict further (does this open up an attack vector ?)
+            // This is required for API call during discovery (to get site name and initialisation status)
```

Comment:
Probably better for client to allow this

Reply:
The user-agent header could be set by the caller so I guess this would not be secure...

Can we allow the request based on some selected routes instead?

Reply:
> The user-agent header could be set by the caller so I guess this would not be secure...

cors protects against request in trusted clients i.e. browser, user-agent cannot be set by JS in browser

> Can we allow the request based on some selected routes instead?

Could make a rest endpoint for this, which would make extracting route easier, can probably extract gaphql query from `req`. I think this should really be done by the `client` (electron or android web view), where we have full control over the headers we are sending, linking this comment to carry over.

Reply:
Yes right user-agent can't be set from the website. But, for example, if we would have something like a 3rd party FE plugin, they could just get a token on behave of the user if they pretend they are an electron client. Or if you allow to show an external link or something. Seems tricky to me. Probably safer to explicitly allow safe endpoints to use without cors.

Reply:
I am not quite sure about this use case, 3rd party FE plugin would be JS code that runs in the context of our web app ? If it's 3rd party and not sandboxed, or not reviewed and signed, then they can do anything, even recording user password during login and send to 3rd party server.

Maybe i misunderstood your use case ?

Since we have control over `client` for electron/android, we it's probably best if they just allow this during discovery (rather then server having edge cases in cors)

Reply:
yes its quite difficult to keep the different threats in mind and I am not 100% sure if the use cases are valid. What brings me to the conclusion that its just to hard to get it correct and the safest way is to just have "standard" CORS handling and only ignore CORS for save endpoints. Even if it is safe now we would have to keep it in mind in the future to not accidentally introduce problems at a later stage..